File: plugins/scandit-sdk/evals/label-capture-android/fixtures/before-8.0-regex.kt
package com.example.app

import com.scandit.datacapture.barcode.data.Symbology
import com.scandit.datacapture.label.capture.LabelCaptureSettings

fun buildLabelCaptureSettings(): LabelCaptureSettings {
    return LabelCaptureSettings.builder()
        .addLabel()
            .addCustomBarcode()
                .setSymbologies(Symbology.EAN13_UPCA)
            .buildFluent("barcode")
            .addExpiryDateText()
                .setDataTypePattern(Regex("EXP[:\\s]+"))
                .setPattern("\\d{2}/\\d{2}/\\d{2,4}")
            .buildFluent("expiry-date")
            .addCustomText()
                .setDataTypePatterns(listOf("LOT[:\\s]+", "Batch[:\\s]+"))
                .setPatterns("[A-Z0-9]{6,}")
                .isOptional(true)
            .buildFluent("lot-number")
        .buildFluent("perishable-product")
        .build()
}

File: plugins/scandit-sdk/evals/label-capture-web/fixtures/before-8.0-regex.ts
import { Symbology } from "@scandit/web-datacapture-barcode";
import {
  CustomBarcodeBuilder,
  CustomTextBuilder,
  ExpiryDateTextBuilder,
  LabelCaptureSettingsBuilder,
  LabelDefinitionBuilder,
} from "@scandit/web-datacapture-label";

export async function buildSettings() {
  const expiryBuilder = new ExpiryDateTextBuilder()
    .setDataTypePattern("EXP[:\\s]+")
    .setPattern("\\d{2}/\\d{2}/\\d{2,4}");

  const lotBuilder = new CustomTextBuilder()
    .setDataTypePatterns(["LOT[:\\s]+", "Batch[:\\s]+"])
    .setPatterns(["[A-Z0-9]{6,}"]);

  return await new LabelCaptureSettingsBuilder()
    .addLabel(
      await new LabelDefinitionBuilder()
        .addCustomBarcode(
          await new CustomBarcodeBuilder()
            .setSymbology(Symbology.EAN13UPCA)
            .build("Barcode")
        )
        .addExpiryDateText(await expiryBuilder.build("Expiry Date"))
        .addCustomText(await lotBuilder.build("Lot Number"))
        .build("Perishable Product")
    )
    .build();
}
